source/Application/Controller/Admin/ActionsGroupsAjax.php

```diff
@@ -124,9 +123,10 @@ public function addPromotionGroup()
                 $oObject2Promotion->oxobject2action__oxclass = new oxField("oxgroups");
                 $oObject2Promotion->save();
             }
-            $promotionAdded = true;
+
+            return true;
```

Comment:
I don't think that this is the case where **early return** should be used, some info can be found here: http://williamdurand.fr/2013/06/03/object-calisthenics/#dont-use-the-else-keyword
Also usually in OXID eShop code it is used one `return` statement per method.